### PR TITLE
Feature/decore class

### PR DESCRIPTION
**Pull request by ShinhyoBelok**
# Changes 🛠️ 

#### Interface
- Create a class `Nameable`.
   - Implement a method called `correct_name` that will raise a `NotImplementedError`.

#### Person class to Nameable
- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

#### base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require './nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super
```
Comment (anagudelogu) [inline base_decorator.rb]: - [ ] Please be extra careful using the super keyword, the default behavior is to call the same method on the parent with the same arguments, but... What if the methods don't share the same number of arguments? 😱 In this case, Nameable doesn't have an explicit initialize method, so by default, it receives no arguments, but here you're calling it with `nameable` as an argument which causes an error, you can read more about the `super` keyword [here](https://www.rubyguides.com/2018/09/ruby-super-keyword/). Please consider adding a `()` to indicate that it doesn't receive any arguments, like this: 🤓 🔧 

```suggestion
    super()
```

File: person.rb [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+require './nameable'
+require './capitalize_decorator'
+require './trimmer_decorator'
+
+class Person < Nameable
   attr_reader :id
   attr_accessor :name, :age
 
   def initialize(age, name = 'unknown', parent_permission: true)
+    super
```
Comment (anagudelogu) [inline person.rb]: - [ ] The same happens here, please consider adding the parenthesis! 🤓 🔧 

```suggestion
    super()
```

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @ShinhyoBelok, 
I'm Andres.

While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @ShinhyoBelok , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] No linter errors :white_check_mark:
- [x] Gitflow is used correctly :white_check_mark:  
- [x]  Good descriptive commit messages :white_check_mark: 
- [x] Implemented all the changes from the previous reviewer :white_check_mark: 
- [x] Descriptive pull request title and summary :white_check_mark:  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**